src/providers/wms/qgswmsprovider.cpp

```diff
@@ -2990,19 +2990,24 @@ QUrl QgsWmsProvider::getLegendGraphicFullURL( double scale, const QgsRectangle&
 
   QUrl url( lurl );
 
-  if ( !url.hasQueryItem( "SERVICE" ) )
+  // query names/keys are NOT case-sensitive, so make an uppercase list for proper comparison
+  QList<QString> qnames;
+  for(int i = 0; i < url.queryItems().size(); i++) {
+    qnames << url.queryItems().at(i).first.toUpper();
```

Comment:
The style check will complain about it..., you should add some space e.g.: `(i)` => `( i )`